### ISSUE TITLE
Add support for logs of all lambda functions

### ISSUE THREAD

**Issue by jamesls** [aws/chalice#841]
The `chalice logs` command was written when there was only a single lambda function used as your API handler.  I've found it really useful, but I'd like to be able to use this command for all the lambda functions in my app.

I'd like a `-f/--function-name` arg added that lets me provide the logical resource name of the function whose logs I'd like.

### Example

Given this app:

```python

@app.lambda_function()
def foo(event, context): ...

@app.route('/')
def index(): ...
```

The `chalice logs` command will continue to give me lambda logs for my API handler, but I can now also say:

```
$ chalice logs -f foo
```

and it will give me logs for the `foo` function.  Also I'd like to get https://github.com/aws/chalice/issues/4 added as well so I might need to work out if we want `-f` to mean what `tail -f` means or if we just want to use `--tail`.

**Comment by kyleknap** [comment]
Yes! Both would be awesome.

**Comment by BigChief45** [comment]
This sounds excellent. Never knew there was a `chalice logs` command. I am assuming these logs get pulled from CloudWatch?

**Comment by jamesls** [comment]
@BigChief45 yep they get pulled from cloudwatch logs.  It also has some abstraction on top of the low level cloudwatch logs call to make the logs easier to read.  I added a few examples over in https://github.com/aws/chalice/issues/841 but our docs could probably use more examples.

@kadrach I'd like that as well, it's sometimes the only way to track down issues with bad responses from Lambda.  I've created https://github.com/aws/chalice/issues/849 to track this work.
